File: cpp/tokoelektronik.cpp
#include <bits/stdc++.h>

using namespace std;

class TokoElektronik{
    private:
        //atribut private 
        string id;
        string nama;
        string kategori;
        string harga;

    public:
        //Konstruktor kosong / default
        TokoElektronik(){
            id = "";
            nama = "";
            kategori = "";
            harga = "";
        }

        //Konstruktor dengan parameter
        TokoElektronik(string id, string nama, string kategori, string harga){ 
            this->id = id;
            this->nama = nama;
            this->kategori = kategori;
            this->harga = harga;
        }

        //setter dan getter
        void setId(string id){
            this->id = id;
        }
        
        string getId(){
            return id;
        }

        void setNama(string nama){
            this->nama = nama;
        }

        string getNama(){
            return nama;
        }

        void setKategori(string kategori){
            this->kategori = kategori;
        }

        string getKategori(){
            return kategori;
        }

        void setHarga(string harga){
            this->harga = harga;
        }

        string getHarga(){
            return harga;
        }

        //method menampilkan data
        void display(){
            cout << "ID       : " << id << endl;
            cout << "Nama     : " << nama << endl;
            cout << "Kategori : " << kategori << endl;
            cout << "Harga    : " << harga << endl;
        }

        //destruktor
        ~TokoElektronik(){

        }
};

File: java/Tokoelektronik.java
public class Tokoelektronik {
    //atribut private
    private String id;
    private String nama;
    private String kategori;
    private String harga;

    //konstruktor default
    public Tokoelektronik() {
        this.id = "";
        this.nama = "";
        this.kategori = "";
        this.harga = "";
    }

    //konstruktor dengan parameter
    public Tokoelektronik(String id, String nama, String kategori, String harga) {
        this.id = id;
        this.nama = nama;
        this.kategori = kategori;
        this.harga = harga;
    }

    //metode setter dan getter
    void setId(String id){
        this.id = id;
    }

    String getId(){
        return id;
    }

    void setNama(String nama){
        this.nama = nama;
    }

    String getNama(){
        return nama;
    }

    void setKategori(String kategori){
        this.kategori = kategori;
    }

    String getKategori(){
        return kategori;
    }

    void setHarga(String harga){
        this.harga = harga;
    }

    String getHarga(){
        return harga;
    }

    void display(){//metode untuk menampilkan data
        System.out.println("ID       : " + id);
        System.out.println("Nama     : " + nama);
        System.out.println("Kategori : " + kategori);
        System.out.println("Harga    : " + harga);
    }
}